### PR TITLE
Allow default retry count to be determined by environment variable

### LINKED ISSUE #675: Allow default retry count to be determined by environment variable
## Description

Most of our DAGs share default arguments, which can be found here: 

https://github.com/WordPress/openverse-catalog/blob/492ae8b521dd2cf7356b1295c68c72feaf7d491c/openverse_catalog/dags/common/constants.py#L14-L23

By default (and unless explicitly stated at the DAG or task level), DAGs which use these arguments have their retries set to `2`, which means that the task will run 3 total times before it eventually fails.

For local development, these retries can be cumbersome and time-consuming. It would be great if we could have that value configured from an environment variable which defaults to `2` if not set.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by Pmeet**
Modified the DAG_DEFAULT_ARGS dictionary's retries record and added the DEFAULT_RETRY_COUNT to 2 in env.template file and reformatted both files according to flake8

## Fixes

Fixes #675  by @AetherUnbound 

## Description

I fixed this issue by adding the environment variable for the retries in env.template and fetching the same variable in constants.py file using os.getenv(<variable-name>). 

## Testing Instructions

Not Applicable

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/common/constants.py [hunk 1]
```diff
@@ -22,7 +22,9 @@
     "email_on_failure": False,
     "email_on_retry": False,
     "email": [CONTACT_EMAIL],
-    "retries": 2,
+    "retries": os.getenv(
+        "DEFAULT_RETRY_COUNT"
+    ),  # Default retries is set to 2 in .env file,
```
Comment (AetherUnbound) [inline openverse_catalog/dags/common/constants.py:27]: In the case where this environment variable isn't set, we'd still like this to default to 2. We can change change that by doing:
```suggestion
    "retries": os.getenv("DEFAULT_RETRY_COUNT", 2),
```

**Review by AetherUnbound — CHANGES_REQUESTED** [review]
Thanks for submitting this PR @Pmeet! I have one suggested change (in some environments we don't specify all env vars or use the `.env` file at all, so we want to make sure we have that case covered).

**Review by AetherUnbound — APPROVED** [review]
This looks great, thanks @Pmeet and congrats on your first contribution! 😄

**Review by stacimc — APPROVED** [review]
Looks great 😄 Thanks for the contribution!

**Pull request merged**